command/setup_consul.go

```diff
@@ -0,0 +1,449 @@
+	// reuse the same connection.
+	client *api.Client
+
+	jwksURL string
+
+	// if set, answers "Yes" to all the interactive questions
+	autoYes bool
+}
+
+// Help satisfies the cli.Command Help function.
+func (s *SetupConsulCommand) Help() string {
+	helpText := `
+Usage: nomad setup consul [options]
+
+  This command sets up Consul for allowing Nomad workloads to authenticate
+  themselves using Workload Identity.
+
+  Setup Consul for Nomad:
+
+	$ nomad setup consul -y -jwks="http://nomad.example/.well-known/jwks.json"
+`
+	return strings.TrimSpace(helpText)
+}
+
+func (s *SetupConsulCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(s.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-jwks-url": complete.PredictAnything,
+			"-y":        complete.PredictSet("true", "false"),
+		})
+}
+
+func (s *SetupConsulCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+// Synopsis satisfies the cli.Command Synopsis function.
+func (s *SetupConsulCommand) Synopsis() string { return "Interact with setup helpers" }
+
+// Name returns the name of this command.
+func (s *SetupConsulCommand) Name() string { return "setup" }
+
+// Run satisfies the cli.Command Run function.
+func (s *SetupConsulCommand) Run(args []string) int {
+
+	flags := s.Meta.FlagSet(s.Name(), FlagSetClient)
+	flags.Usage = func() { s.Ui.Output(s.Help()) }
+	flags.BoolVar(&s.autoYes, "y", false, "")
+	flags.StringVar(&s.jwksURL, "jwks", "http://localhost:4646/.well-known/jwks.json", "")
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	// Check that we got no arguments.
+	if len(flags.Args()) != 0 {
+		s.Ui.Error("This command takes no arguments")
+		s.Ui.Error(commandErrorText(s))
+		return 1
+	}
+
+	var err error
+
+	// Get the Consul client.
+	cfg := api.DefaultConfig()
+	s.client, err = api.NewClient(cfg)
+	if err != nil {
+		s.Ui.Error(fmt.Sprintf("Error initializing Consul client: %s", err))
+		return 1
+	}
+
+	s.Ui.Output(`
+This command will walk you through configuring all the components required for 
+Nomad workloads to authenticate themselves against Consul ACL using their 
+respective Workload Identities. 
+
+First we need to create a JWT auth method for Nomad services. Here is the auth
+method configuration we will create:
+`)
+
+	authMethodConf, err := s.renderAuthMethodConf(authMethodServices)
+	if err != nil {
+		s.Ui.Error(err.Error())
+		return 1
+	}
+
+	jsConf, _ := json.MarshalIndent(authMethodConf, "", "\t")
+	s.Ui.Output(string(jsConf))
+
+	var createAuthMethod bool
+	if !s.autoYes {
+		createAuthMethod = s.askQuestion(
+			fmt.Sprintf(
+				"Should we create the %s auth method in your Consul cluster? [Y/n]",
+				authMethodServices,
+			))
+	} else {
+		createAuthMethod = true
+	}
+
+	if createAuthMethod {
+		err = s.createAuthMethod(authMethodServices, authMethodConf)
+		if err != nil {
+			s.Ui.Error(err.Error())
+			return 1
+		}
+	}
```

Comment:
My reasoning was: if users reply `n` this indicates they either set some parts up themselves or prefer to set them up themselves. I think otherwise it makes no sense to have the interactive helper at all.